### PR TITLE
NAVAND-547 route refresh for new annotations

### PR DESCRIPTION
**Pull request by VysotskiVadim**
### Description
Companion PR for https://github.com/mapbox/mapbox-java/pull/1567

### CODE REVIEW

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/AnnotationsRefresher.kt [hunk 1]
```diff
@@ -48,6 +48,16 @@ internal object AnnotationsRefresher {
             newAnnotation,
             startingLegGeometryIndex,
         ) { maxspeed() }
+        val freeFlowSpeed = mergeAnnotationProperty(
```
Comment (dzinad) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/AnnotationsRefresher.kt:51]: Some changes are also required in `DirectionsRouteDiffProvider`.

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteAlternativesTest.kt [hunk 2]
```diff
@@ -144,6 +147,19 @@ class RouteAlternativesTest : BaseCoreNoCleanUpTest() {
...
             }
+
+            val mockedAlternativesResponse = InputStreamReader(
+                openRawResource(context, R.raw.route_response_alternative_continue)
+            ).use {
+                DirectionsResponse.fromJson(it)
+            }
+            newAlternatives.forEach {
+                assertEquals(
+                    "some info was lost during NN -> Nav SDK transition",
```
Comment (dzinad) [inline instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteAlternativesTest.kt:158]: ?

File: libnavigation-base/src/test/java/com/mapbox/navigation/base/internal/route/AnnotationsRefresherTest.kt [hunk 3]
```diff
@@ -219,6 +219,8 @@ class AnnotationsRefresherTest(
                         .distance(listOf(1.2, 3.4, 5.6, 7.8, 9.0))
                         .duration(listOf(11.2, 33.4, 55.6, 77.8, 99.0))
                         .speed(listOf(41.0, 42.5, 43.1, 44.6, 45.9))
+                        .freeflowSpeed(listOf(1, 2, 3, 4, 5))
```
Comment (dzinad) [inline libnavigation-base/src/test/java/com/mapbox/navigation/base/internal/route/AnnotationsRefresherTest.kt:222]: Are you also gonna test it in instrumentation tests? I see a lot of room for error for NN team, they already said something would be wrong after refresh.

Comment (VysotskiVadim) [reply libnavigation-base/src/test/java/com/mapbox/navigation/base/internal/route/AnnotationsRefresherTest.kt:222]: Which test do you suggest?

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteAlternativesTest.kt (hunk 2 above)
Comment (VysotskiVadim) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteAlternativesTest.kt:158]: it means that info from response doesn't match info we get from NN

File: libnavigation-base/src/test/java/com/mapbox/navigation/base/internal/route/AnnotationsRefresherTest.kt (hunk 3 above)
Comment (dzinad) [reply libnavigation-base/src/test/java/com/mapbox/navigation/base/internal/route/AnnotationsRefresherTest.kt:222]: The same, but with NN involved. :)
We already have some tests that test annotations refresh, we could just extend them.

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteAlternativesTest.kt (hunk 2 above)
Comment (dzinad) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteAlternativesTest.kt:158]: What does it have to do with the new annotation?

**Review by dzinad — APPROVED** [review]

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/AnnotationsRefresher.kt (hunk 1 above)
Comment (VysotskiVadim) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/AnnotationsRefresher.kt:51]: done

**Pull request merged**